wherehows-frontend/app/controllers/api/v2/Dataset.java

```diff
@@ -246,6 +250,46 @@ private static AclDao initAclDao() {
     return Promise.promise(() -> ok(Json.newObject().set("health", Json.toJson(health))));
   }
 
+  public static Promise<Result> getExportPolicy(String datasetUrn) {
+    final DatasetExportPolicy exportPolicy;
+    try {
+      exportPolicy = EXPORT_POLICY_DAO.getDatasetExportPolicy(datasetUrn);
+    } catch (Exception e) {
+      if (e.toString().contains("Response status 404")) {
+        return Promise.promise(() -> notFound(_EMPTY_RESPONSE));
+      }
+
+      Logger.error("Fetch export policy fail", e);
+      return Promise.promise(() -> internalServerError(errorResponse(e)));
+    }
+
+    if (exportPolicy == null) {
+      return Promise.promise(() -> notFound(_EMPTY_RESPONSE));
+    }
+
+    return Promise.promise(() -> ok(Json.newObject().set("exportPolicy", Json.toJson(exportPolicy))));
+  }
+
+  public static Promise<Result> updateExportPolicy(String datasetUrn) {
+    final String username = session("user");
+    final DatasetExportPolicy exportPolicy;
+
+    if (StringUtils.isBlank(username)) {
+      return Promise.promise(() -> unauthorized(_EMPTY_RESPONSE));
+    }
+
+    try {
+      exportPolicy = Json.mapper().convertValue(request().body().asJson(), DatasetExportPolicy.class);
```

Comment:
Before "try", can add   JsonNode json = null;
And at line 282,  can save  request().body().asJson() to this variable
And in line 290, can reuse this variable in response, so don't need to cast java object back to Json again.